src/app/services/transmart-resource/transmart-mapper.ts

```diff
@@ -131,9 +132,9 @@ export class TransmartMapper {
         transmartColumnHeader.elements.forEach(elem => {
           let metadata = this.getDimensionMetadata(transmartColumnHeader.dimension, elem);
           if (isUsingHeaders) {
-            this.updateCols(headerRow.cols, elem['label'], metadata);
+            this.updateCols(headerRow.cols, elem[headerNameField], metadata);
           } else {
-            row.addDatum(elem['label'], metadata);
+            row.addDatum(elem[headerNameField], metadata);
```

Comment:
I never heard of singular of Data used as Datum. That is interesting. Though not related to the PR.